.github/workflows/format.yml

```diff
@@ -0,0 +1,24 @@
+name: Scalafmt
+
+permissions: {}
+
+on:
+  pull_request:
+    branches: ['**']
+
+jobs:
+  build:
+    name: Code is formatted
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout current branch (full)
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          persist-credentials: false
+
+      - name: Check project is formatted
+        uses: jrouly/scalafmt-native-action@v2
+        with:
+          version: '3.7.3'
+          arguments: '--list --mode diff-ref=origin/main'
```

Comment:
This means that scalafmt will only check files that are have been changed from `origin/main` according to git.